discord/commands/options.py

```diff
@@ -392,11 +388,31 @@ def __repr__(self):
         return f"<discord.commands.{self.__class__.__name__} name={self.name}>"
 
     @property
-    def autocomplete(self) -> Callable[[Any], Any, Any] | None:
+    def autocomplete(self):
         return self._autocomplete
 
     @autocomplete.setter
-    def autocomplete(self, value: Callable[[Any], Any, Any] | None) -> None:
+    def autocomplete(self, value) -> None:
+        """
+        The autocomplete handler for the option. Accepts a callable (sync or async)
+        that takes a single required argument of :class:`AutocompleteContext`.
+        The callable must return an iterable of :class:`str` or :class:`OptionChoice`.
+        Alternatively, :func:`discord.utils.basic_autocomplete` may be used in place of the callable.
+
+        Parameters
+        ----------
+        value: Union[
+            Callable[[Self, AutocompleteContext, Any], AutocompleteReturnType],
+            Callable[[AutocompleteContext, Any], AutocompleteReturnType],
+            Callable[[Self, AutocompleteContext, Any], Awaitable[AutocompleteReturnType]],
```

Comment:
To be exact, this typing would need protocols but it's not worth the effort right here